jpo-asn-runtime/src/main/java/us/dot/its/jpo/asn/runtime/types/Asn1Integer.java

```diff
@@ -1,66 +1,83 @@
 package us.dot.its.jpo.asn.runtime.types;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonValue;
 import java.util.Objects;
+import java.util.Optional;
+import lombok.Getter;
+import lombok.Setter;
 
 /**
- * Represents an ASN.1 INTEGER type, and is also used as a base class for INTEGER type
- * assignments with or without size constraints.
+ * Represents an ASN.1 INTEGER type, and is also used as a base class for INTEGER type assignments
+ * with or without size constraints.
  */
 public class Asn1Integer implements Asn1Type, Comparable<Asn1Integer> {
 
-    protected long value;
-    final long lowerBound;
-    final long upperBound;
+  @Setter
+  protected long value;
+  @Getter
+  final long lowerBound;
+  @Getter
+  final long upperBound;
 
-    public Asn1Integer(long lowerBound, long upperBound) {
-        this.lowerBound = lowerBound;
-        this.upperBound = upperBound;
-    }
+  public Asn1Integer() {
+    this(Long.MIN_VALUE, Long.MAX_VALUE);
+  }
 
-    @JsonValue
-    public long getValue() {
-        return value;
-    }
+  @JsonCreator
+  public Asn1Integer(long value) {
+    this();
+    this.value = value;
+  }
 
-    public void setValue(long value) {
-        this.value = value;
-    }
+  public Asn1Integer(long lowerBound, long upperBound) {
+    this.lowerBound = lowerBound;
+    this.upperBound = upperBound;
+  }
 
-    public int intValue() {
-        return (int)value;
-    }
+  @JsonValue
+  public long getValue() {
+    return value;
+  }
 
-    @Override
-    public int compareTo(Asn1Integer other) {
-        if (other == null) return -1;
-        return Long.compare(value, other.value);
-    }
+  public int intValue() {
+    return (int) value;
+  }
 
-    public long getLowerBound() {
-        return lowerBound;
+  @Override
+  public int compareTo(Asn1Integer other) {
+    if (other == null) {
+      return -1;
     }
+    return Long.compare(value, other.value);
+  }
+
 
-    public long getUpperBound() {
-        return upperBound;
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
     }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    Asn1Integer that = (Asn1Integer) o;
+    return value == that.value;
+  }
 
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(value);
+  }
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-        Asn1Integer that = (Asn1Integer) o;
-        return value == that.value;
-    }
+  @Override
+  public String toString() {
+    return Long.toString(value);
+  }
+
+  public Optional<String> name() {

```

Comment:
question: what's this used for if we always return Optional.empty?

Reply:
It is the default in the base class for most integer types that don't have named values, if you ask for the name, it returns nothing.  But it is overridden for ints with named values to return the name if there is one.